rs/ic_os/config/src/types.rs

```diff
@@ -83,12 +84,16 @@ pub struct BackupSpoolSettings {
 
 #[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
 pub struct ICOSSettings {
+    /// in nested testing, mgmt_mac is set in deployment.json.template,
+    /// else found dynamically in call to config tool CreateSetuposConfig
+    pub mgmt_mac: FormattedMacAddress,
+    /// "mainnet" or "testnet"
+    pub deployment_environment: String,
     pub logging: Logging,
     /// This file must be a text file containing the public key of the NNS to be used.
     pub nns_public_key_path: PathBuf,
     /// The URL (HTTP) of the NNS node(s).
     pub nns_urls: Vec<Url>,
-    pub hostname: String,
```

Comment:
Renamed "hostname" to "deployment_environment". It was never actually the hostname and rather storing "mainnet" or "testnet"